mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -6,15 +6,79 @@
 from typing import TYPE_CHECKING
 from pathlib import Path
 from logging import getLogger
+
+import numpy as np
 from PyQt5.QtCore import QFileSystemWatcher, QObject, pyqtSignal, QTimer
 
+from tifffile import tifffile
+from astropy.io import fits
+
+from mantidimaging.core.utility.sensible_roi import SensibleROI
+
 if TYPE_CHECKING:
     from os import stat_result
     from mantidimaging.gui.windows.live_viewer.view import LiveViewerWindowPresenter
 
 LOG = getLogger(__name__)
 
 
+def load_image_from_path(image_path: Path) -> np.ndarray:
+    """
+    Load a .Tif, .Tiff or .Fits file only if it exists
+    and returns as an ndarray
+    """
+    if image_path.suffix.lower() in [".tif", ".tiff"]:
+        with tifffile.TiffFile(image_path) as tif:
+            image_data = tif.asarray()
+    elif image_path.suffix.lower() == ".fits":
+        with fits.open(image_path.__str__()) as fit:
+            image_data = fit[0].data
+    return image_data
+
+
+class ImageCache:
```

Comment:
All the methods in the class should have type annotations for return values.

And I don't think `load_image()` ever returns `None`.

Reply:
Also, as `load_image()` is the only method that we expect to be used from outside the class, we could mark the others as 'private' with a leading underscore, e.g. `_add_to_cache()`

Reply:
`load_image` can return `None` if an exception occurs, it doesn't reach a return statement

Reply:
If it raises an exception, then it does not count as a return, so `None` is not needed in the annotation.